test/integration/models/test_linode.py

```diff
@@ -0,0 +1,431 @@
+
+    wait_for_condition(10, 100, get_status, linode, "resizing")
+
+    assert linode.status == "resizing"
+
+    # Takes about 3-5 minute to resize, sometimes longer...
+    wait_for_condition(30, 600, get_status, linode, "running")
+
+    assert linode.status == "running"
+
+
+def test_linode_resize_with_class(
+    get_client, create_linode_for_long_running_tests
+):
+    linode = create_linode_for_long_running_tests
+    ltype = Type(get_client, "g6-standard-6")
+
+    wait_for_condition(10, 100, get_status, linode, "running")
+
+    time.sleep(5)
+    res = linode.resize(new_type=ltype)
+
+    assert res
+
+    wait_for_condition(10, 300, get_status, linode, "resizing")
+
+    assert linode.status == "resizing"
+
+    # Takes about 3-5 minute to resize, sometimes longer...
+    wait_for_condition(30, 600, get_status, linode, "running")
+
+    assert linode.status == "running"
+
+
+def test_linode_boot_with_config(create_linode):
+    linode = create_linode
+
+    wait_for_condition(10, 100, get_status, linode, "running")
+    retry_sending_request(3, linode.shutdown)
+
+    wait_for_condition(30, 300, get_status, linode, "offline")
+
+    config = linode.configs[0]
+
+    retry_sending_request(3, linode.boot, config)
+
+    wait_for_condition(10, 100, get_status, linode, "running")
+
+    assert linode.status == "running"
+
+
+def test_linode_firewalls(create_linode_with_volume_firewall):
+    linode = create_linode_with_volume_firewall
+
+    firewalls = linode.firewalls()
+
+    assert len(firewalls) > 0
+    assert "TestSDK" in firewalls[0].label
+
+
+def test_linode_volumes(create_linode_with_volume_firewall):
+    linode = create_linode_with_volume_firewall
+
+    volumes = linode.volumes()
+
+    assert len(volumes) > 0
+    assert "TestSDK" in volumes[0].label
+
+
+def test_linode_disk_duplicate(get_client, create_linode):
+    pytest.skip("Need to find out the space sizing when duplicating disks")
+    linode = create_linode
+
+    disk = get_client.load(Disk, linode.disks[0].id, linode.id)
+
+    try:
+        dup_disk = disk.duplicate()
+        assert dup_disk.linode_id == linode.id
+    except ApiError as e:
+        assert e.status == 400
+        assert "Insufficient space" in str(e.json)
+
+
+def test_linode_instance_password(create_linode_for_pass_reset):
+    pytest.skip("Failing due to mismatched request body")
+    linode = create_linode_for_pass_reset[0]
+    password = create_linode_for_pass_reset[1]
+
+    wait_for_condition(10, 100, get_status, linode, "running")
+
+    retry_sending_request(3, linode.shutdown)
+
+    wait_for_condition(10, 200, get_status, linode, "offline")
+
+    linode.reset_instance_root_password(root_password=password)
+
+    linode.boot()
+
+    wait_for_condition(10, 100, get_status, linode, "running")
+
+    assert linode.status == "running"
+
+
+def test_linode_ips(create_linode):
+    linode = create_linode
+
+    ips = linode.ips
+
+    assert ips.ipv4 == linode.ipv4
```

Comment:
Looks like this line is comparing a dict (`ips.ipv4`; constructed [here](https://github.com/linode/linode_api4-python/blob/dev/linode_api4/objects/linode.py#LL438C17-L438C17)) with a list of strings, which is causing a failure. 

Maybe something like this would work?

```python
assert ips.ipv4.public[0].address in linode.ipv4
```

Reply:
Thanks for the catch